homeassistant/components/airthings/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""Support for Airthings sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from airthings import AirthingsDevice
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+    SensorEntityDescription,
+    StateType,
+)
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_BILLION,
+    CONCENTRATION_PARTS_PER_MILLION,
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_CO2,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_PRESSURE,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS,
+    PERCENTAGE,
+    PRESSURE_MBAR,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+
+@dataclass
+class AirthingsSensorEntityDescription(SensorEntityDescription):
+    """Describes Airthings sensor entity."""
+
+    sensor_name: str | None = None
+
+
+SENSORS: dict[str, AirthingsSensorEntityDescription] = {
+    "radonShortTermAvg": AirthingsSensorEntityDescription(
+        key="radonShortTermAvg",
+        native_unit_of_measurement="Bq/m³",
+        sensor_name="Radon",
+    ),
+    "temp": AirthingsSensorEntityDescription(
+        key="temp",
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        sensor_name="Temperature",
+    ),
+    "humidity": AirthingsSensorEntityDescription(
+        key="humidity",
+        device_class=DEVICE_CLASS_HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        sensor_name="Humidity",
+    ),
+    "pressure": AirthingsSensorEntityDescription(
+        key="pressure",
+        device_class=DEVICE_CLASS_PRESSURE,
+        native_unit_of_measurement=PRESSURE_MBAR,
+        sensor_name="Pressure",
+    ),
+    "battery": AirthingsSensorEntityDescription(
+        key="battery",
+        device_class=DEVICE_CLASS_BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        sensor_name="Battery",
+    ),
+    "co2": AirthingsSensorEntityDescription(
+        key="co2",
+        device_class=DEVICE_CLASS_CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        sensor_name="CO2",
+    ),
+    "voc": AirthingsSensorEntityDescription(
+        key="voc",
+        device_class=DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
```

Comment:
Currently we only allow mikrogram/cubic meter. This may change if we see more integrations requiring a different unit but for now we can't use this device class unless we use the unit in the docs.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes